### PR TITLE
[FL-2759], [FL-2766] NFC collect params for mfkey32 attack

### PR DESCRIPTION
**Pull request by gornekich**
# What's new

- Add Reader Analyzer logic
- Add mfkey parameters collection on SD card in Detect Reader menu
- Add plain text logs saved on SD card while reading NFC tags
- Fix ISO-14443-4 UID emulation

# Verification 
- Nothing is broken
- ISO-14443-4 UID emulation is fixed
- Collect nonced from reader with Detect Reader option. Run `mfkey_offline.py` from https://github.com/flipperdevices/flipperzero-nfc-rpc
- Set Debug ON. Read cards. Run `read_log.py` from https://github.com/flipperdevices/flipperzero-nfc-rpc

# Checklist (For Reviewer)

- [x] Description contains actions to verify feature/bugfix
- [x] PR has description of feature/bug or link to Confluence/Jira task
- [x] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

**Comment by RaitoBezarius** [comment]
I tested the compiled firmware for UID emulation with Mifare Classic 1k for example, and it did not work, using commit 73ff061.
Reproducer: "Add Manually" > "Mifare Classic (1|4)K (7|4) Bytes" > "Emulate", use whatever read (I used my phone in that case.)

But I can confirm the reader analyze produce valid Mifare Classic emulation.

File: lib/nfc/nfc_worker.h [hunk 1]
```diff
@@ -16,6 +16,7 @@ typedef enum {
     NfcWorkerStateMfClassicEmulate,
     NfcWorkerStateReadMfUltralightReadAuth,
     NfcWorkerStateMfClassicDictAttack,
+    NfcWorkerStateAnalizeReader,
```
Comment (RaitoBezarius) [inline lib/nfc/nfc_worker.h]: Analyze you meant here I think

Comment (gornekich) [reply lib/nfc/nfc_worker.h]: thx

**Comment by gornekich** [comment]
Hello @RaitoBezarius . Can't reproduce your issue. I use proxmark and another flipper, UID emulation works just fine. Are you sure that your mobile application reads just UID, not data inside?

**Comment by RaitoBezarius** [comment]
> Hello @RaitoBezarius . Can't reproduce your issue. I use proxmark and another flipper, UID emulation works just fine. Are you sure that your mobile application reads just UID, not data inside?

Yeah, it's Mifare Classic Tool in Display Tag Info which reads only the UID and not the data inside, I have been trying with multiple UIDs and still not work. Which menu / usecase did you try?

**Comment by Alejandro12120** [comment]
Hey @gornekich, I have tried your PR but mfkey doesn't return any key. This is my mfkey32.log:

`Sector 2 key A cuid 2a234f80 nt0 4fd6a74d nr0 c3633368 ar0 c2eab3c2 nt1 ee1cafea nr1 36e54ad9 ar1 2457cd2b`

**Comment by gornekich** [comment]
Helllo @Alejandro12120 . Looks like you need to collect more parameters and try to run mfkey32v2 again. Sometimes attack fails

**Comment by gornekich** [comment]
> > Hello @RaitoBezarius . Can't reproduce your issue. I use proxmark and another flipper, UID emulation works just fine. Are you sure that your mobile application reads just UID, not data inside?
> 
> Yeah, it's Mifare Classic Tool in Display Tag Info which reads only the UID and not the data inside, I have been trying with multiple UIDs and still not work. Which menu / usecase did you try?

I tried with Display Tag Info and it's still works fine. What log level in System->Settings do you use? Try to set Default

**Comment by Alejandro12120** [comment]
> Helllo @Alejandro12120 . Looks like you need to collect more parameters and try to run mfkey32v2 again. Sometimes attack fails

But as far as I understood you have to emulate the card that works on that reader, you can't emulate whatever card right?
Maybe you should make a scene where you scan the card and then FZ automatically emulates it and starts collecting nonces.

**Review by skotopes — APPROVED** [review]

**Comment by gornekich** [comment]
> > Helllo @Alejandro12120 . Looks like you need to collect more parameters and try to run mfkey32v2 again. Sometimes attack fails
> 
> 
> 
> But as far as I understood you have to emulate the card that works on that reader, you can't emulate whatever card right?
> 
> Maybe you should make a scene where you scan the card and then FZ automatically emulates it and starts collecting nonces.
> 
> 

Yes, we want to add configure option on detect reader screen, where user can choose card or pre-scan. Will come in next iteration

**Comment by Alejandro12120** [comment]
> > > Helllo @Alejandro12120 . Looks like you need to collect more parameters and try to run mfkey32v2 again. Sometimes attack fails
> > 
> > 
> > 
> > But as far as I understood you have to emulate the card that works on that reader, you can't emulate whatever card right?
> > 
> > Maybe you should make a scene where you scan the card and then FZ automatically emulates it and starts collecting nonces.
> > 
> > 
> 
> Yes, we want to add configure option on detect reader screen, where user can choose card or pre-scan. Will come in next iteration

Thanks for your reply, do you have an ETA for the next iteration?

**Comment by gornekich** [comment]
> > > > Helllo @Alejandro12120 . Looks like you need to collect more parameters and try to run mfkey32v2 again. Sometimes attack fails
> > > 
> > > 
> > > But as far as I understood you have to emulate the card that works on that reader, you can't emulate whatever card right?
> > > Maybe you should make a scene where you scan the card and then FZ automatically emulates it and starts collecting nonces.
> > 
> > 
> > Yes, we want to add configure option on detect reader screen, where user can choose card or pre-scan. Will come in next iteration
> 
> Thanks for your reply, do you have an ETA for the next iteration?

I think end of september

**Pull request merged**